src/unitxt/fusion.py

```diff
@@ -44,40 +45,59 @@ class FixedFusion(BaseFusion):
     """FixedFusion operator that combines multiple streams into one based on a fixed number of examples per task.
 
     Args:
-        origins: List of SourceOperator objects.
+        origins: Dict of named SourceOperator objects.
         examples_per_task: Number of examples per task. If None, all examples are returned.
         splits: List of splits to include. If None, all splits are included.
     """
 
     max_instances_per_origin: Optional[int] = None
 
+    # flake8: noqa: C901
     def fusion_generator(self, split) -> Generator:
-        for origin in self.origins:
+        for origin_name, origin in self.origins.items():
             multi_stream = origin()
             if split not in multi_stream:
                 continue
             iterator = iter(multi_stream[split])
             if self.max_instances_per_origin is not None:
                 for _ in range(self.max_instances_per_origin):
                     try:
-                        yield next(iterator)
+                        instance = next(iterator)
+                        if "group" in instance:
+                            if "past_groups" in instance:
+                                instance["past_groups"] = (
+                                    instance["group"] + "/" + instance["past_groups"]
+                                )
+                            else:
+                                instance["past_groups"] = instance["group"]
+                        instance["group"] = origin_name
```

Comment:
```suggestion
                            group = origin_name + "/" + instance["group"]
                        else:
                            group = origin_name 
                        instance["group"] = group
```
Why not like this? more simple and dont require additional variable

Reply:
I thought that code that reads the "group" (e.g, the splitter) will not need to check if any past is glued to this group, and if glued -- peel it off.   Only if the consuming code is interested in the past, it will look into "past_groups", and do whatever it wants.

Reply:
Changed per your request, @elronbandel. We will know what to do if the past haunts us..